staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -404,6 +410,10 @@ func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.L
 }
 
 func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (string, error) {
+	if name, found := service.Annotations[ServiceAnnotationLoadBalancerPIPName]; found && name != "" {
+		return name, nil
+	}
+
```

Comment:
When users set a wrong annotation, we should report an error instead of creating a new PIP silently.

So besides this, we also need to get the PIP and return errors when it couldn't be found.